File: docs/vsto/codesnippet/VisualBasic/ContentControlTemplateWalkthrough/ThisDocument.vb

Public Class ThisDocument

    '<Snippet1>
    Private GroupControl1 As Microsoft.Office.Tools.Word.GroupContentControl
    Private BuildingBlockControl1 As Microsoft.Office.Tools.Word.BuildingBlockGalleryContentControl
    Private BuildingBlockControl2 As Microsoft.Office.Tools.Word.BuildingBlockGalleryContentControl
    '</Snippet1>

    Private Sub ThisDocument_Startup(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Startup
        Try

            '<Snippet2>
            ComboBoxContentControl1.PlaceholderText = "Choose a title, or enter your own"
            ComboBoxContentControl1.DropDownListEntries.Add("Engineer", "Engineer", 0)
            ComboBoxContentControl1.DropDownListEntries.Add("Designer", "Designer", 1)
            ComboBoxContentControl1.DropDownListEntries.Add("Manager", "Manager", 2)

            DropDownListContentControl1.PlaceholderText = _
                "Choose a rating (1 lowest, 3 highest)"
            DropDownListContentControl1.DropDownListEntries.Add("1", "1", 0)
            DropDownListContentControl1.DropDownListEntries.Add("2", "2", 1)
            DropDownListContentControl1.DropDownListEntries.Add("3", "3", 2)
            '</Snippet2>

            '<Snippet3>
            Me.Tables(1).Select()
            GroupControl1 = Me.Controls.AddGroupContentControl("groupControl1")
            '</Snippet3>

            '<Snippet4>
            Dim template1 As Word.Template = TryCast(Me.AttachedTemplate, Word.Template)
            If template1 IsNot Nothing Then

                template1.BuildingBlockEntries.Add("Employee Table", _
                    Word.WdBuildingBlockTypes.wdTypeCustom1, "Employee and Customer Information", _
                    Me.Tables(1).Range, InsertOptions:=Word.WdDocPartInsertOptions.wdInsertContent)
                template1.BuildingBlockEntries.Add("Customer Table", _
                    Word.WdBuildingBlockTypes.wdTypeCustom1, "Employee and Customer Information", _
                    Me.Tables(2).Range, InsertOptions:=Word.WdDocPartInsertOptions.wdInsertContent)
            End If
            '</Snippet4>

            '<Snippet5>
            If Me.FormsDesign = False Then
                Me.ToggleFormsDesign()
            End If
            Me.Tables(2).Delete()
            Me.Tables(1).Delete()
            Me.ToggleFormsDesign()
            '</Snippet5>

            '<Snippet6>
            BuildingBlockControl1 = Me.Controls.AddBuildingBlockGalleryContentControl( _
                Me.Paragraphs(1).Range, "buildingBlockControl1")
            BuildingBlockControl1.BuildingBlockCategory = "Employee and Customer Information"
            BuildingBlockControl1.BuildingBlockType = Word.WdBuildingBlockTypes.wdTypeCustom1
            BuildingBlockControl1.PlaceholderText = "Choose your first building block"

            BuildingBlockControl2 = Me.Controls.AddBuildingBlockGalleryContentControl( _
                Me.Paragraphs(2).Range, "buildingBlockControl2")
            BuildingBlockControl2.BuildingBlockCategory = "Employee and Customer Information"
            BuildingBlockControl2.BuildingBlockType = Word.WdBuildingBlockTypes.wdTypeCustom1
            BuildingBlockControl2.PlaceholderText = "Choose your second building block"
            '</Snippet6>

        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
    End Sub
    Private Sub ThisDocument_Shutdown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shutdown

    End Sub

End Class

File: docs/vsto/codesnippet/CSharp/ContentControlTemplateWalkthrough/ThisDocument.cs
using System;
using System.Data;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Word = Microsoft.Office.Interop.Word;
using Office = Microsoft.Office.Core;

namespace ContentControlTemplateWalkthrough
{
    public partial class ThisDocument
    {
        //<Snippet1>
        private Microsoft.Office.Tools.Word.GroupContentControl groupControl1;
        private Microsoft.Office.Tools.Word.BuildingBlockGalleryContentControl buildingBlockControl1;
        private Microsoft.Office.Tools.Word.BuildingBlockGalleryContentControl buildingBlockControl2;
        //</Snippet1>

        private void ThisDocument_Startup(object sender, System.EventArgs e)
        {
            try
            {
                //<Snippet2>
                comboBoxContentControl1.PlaceholderText = "Choose a title, or enter your own";
                comboBoxContentControl1.DropDownListEntries.Add("Engineer", "Engineer", 0);
                comboBoxContentControl1.DropDownListEntries.Add("Designer", "Designer", 1);
                comboBoxContentControl1.DropDownListEntries.Add("Manager", "Manager", 2);

                dropDownListContentControl1.PlaceholderText =
                    "Choose a rating (1 lowest, 3 highest)";
                dropDownListContentControl1.DropDownListEntries.Add("1", "1", 0);
                dropDownListContentControl1.DropDownListEntries.Add("2", "2", 1);
                dropDownListContentControl1.DropDownListEntries.Add("3", "3", 2);
                //</Snippet2>

                //<Snippet3>
                this.Tables[1].Range.Select();
                groupControl1 = this.Controls.AddGroupContentControl("groupControl1");
                //</Snippet3>

                //<Snippet4>
                Word.Template template1 = this.AttachedTemplate as Word.Template;

                if (template1 != null)
                {
                    object description = null;
                    template1.BuildingBlockEntries.Add("Employee Table",
                        Word.WdBuildingBlockTypes.wdTypeCustom1, "Employee and Customer Information",
                        this.Tables[1].Range, ref description, Word.WdDocPartInsertOptions.wdInsertContent);
                    template1.BuildingBlockEntries.Add("Customer Table",
                        Word.WdBuildingBlockTypes.wdTypeCustom1, "Employee and Customer Information",
                        this.Tables[2].Range, ref description, Word.WdDocPartInsertOptions.wdInsertContent);
                }
                //</Snippet4>

                //<Snippet5>
                if (!this.FormsDesign)
                {
                    this.ToggleFormsDesign();
                }
                this.Tables[2].Delete();
                this.Tables[1].Delete();
                this.ToggleFormsDesign();
                //</Snippet5>

                //<Snippet6>
                buildingBlockControl1 = this.Controls.AddBuildingBlockGalleryContentControl(
                    this.Paragraphs[1].Range, "buildingBlockControl1");
                buildingBlockControl1.BuildingBlockCategory = "Employee and Customer Information";
                buildingBlockControl1.BuildingBlockType = Word.WdBuildingBlockTypes.wdTypeCustom1;
                buildingBlockControl1.PlaceholderText = "Choose your first building block";

                buildingBlockControl2 = this.Controls.AddBuildingBlockGalleryContentControl(
                    this.Paragraphs[2].Range, "buildingBlockControl2");
                buildingBlockControl2.BuildingBlockCategory = "Employee and Customer Information";
                buildingBlockControl2.BuildingBlockType = Word.WdBuildingBlockTypes.wdTypeCustom1;
                buildingBlockControl2.PlaceholderText = "Choose your second building block";
                //</Snippet6>
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
        }

        private void ThisDocument_Shutdown(object sender, System.EventArgs e)
        {
        }

        #region VSTO Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisDocument_Startup);
            this.Shutdown += new System.EventHandler(ThisDocument_Shutdown);
        }

        #endregion
    }
}
